### ISSUE TITLE
OLH Theme Navigation fixes

### ISSUE THREAD

**Issue by StephDriver** [openlibhums/janeway#4225]
From screenreader testing (#4194), looking at the narrow screen navigation the following issues were identified:

- [ ] Sample 1, item 2. hamburger menu is incorrectly identified as "[article number] link". Menu can be entered, which opens up the drop down options but the next tab is to the search icon (magnifying glass) and not into the dropdown.
- [ ] Sample 1, item 10. "button" read next (hamburger menu below author line). no identity. Can be activated but next tab is to the word 'options' which has not been related to the button.
- [ ] Sample 4, item 1. the navigation menu itself doesn't work appropriately when navigating by link, particularly having entered a dropdown, you can fall off the bottom of the menu and end up either back in the main article or back on the main part of the menu - but the screen still showing the drop down, and then not be able to get back. The back button at the top of a dropdown also is not selectably having navigated away from it - e.g. having gone down the list from within the dropdown, and then going back up, instead of reaching' back' one ends up in the main navigation on the 'home item.' In short: navigation menu needs reviewing and updating to work with screenreader navigating by links and/or tabs, both backwards and forwards.
- [ ] Sample Other, item 1. When navigating away from Sample 1 page, e.g. following a link to another page, the back button ends up returning to the article search page (from which this article was chosen) and not to the article itself. This is highly confusing.
- [ ] Sample Other, item 8. navigation menu - Login and Register appear side by side, but all other links have their own line
- [ ] Sample Other, item 23. navigation within pages - a skip to main content would be nice and something that makes it clear one has reached the footer. For long lists of articles, some way to know how long the list - either pagination or a 'showing results x of y would be helpful.

**Comment by joemull** [comment]
Refinement: Because these fixes are not as essential in our a11y upgrade plan as the clean theme fixes, we should only do as much as we can in the time allowed under an XS issue.

**Comment by StephDriver** [comment]
This is looking like a bigger issue - some of it overlapped with button work, and I'm putting my observations here for later:

1. olh nav is not keyboard accessible, not all items in each dropdown may be navigated to by keyboard, e.g. in the "Articles" menu, tabbing from item 1 (Articles) reaches item 2 (special collections) and bypasses item 2 (issues).
2. missing aria markup - for collapsed/expanded
3. screen reader descriptions are unhelpful ( I think this is due to html structure ratehr than specific aria markup - it reads out total numbers of items, but not where you are within that, I think the whole screen-reader nav experience needs reviewing.

Note: As part of button work (#4854) I have added in javascript to allow screenreader users to know popup states (collapsed/expanded) for the narrow screen menu and search toggle buttons - this may be reusable for the nav dropdowns.
